genesis/sensors/data_handlers.py

```diff
@@ -189,11 +196,31 @@ def initialize(self):
         self.file_handle = open(self.filename, "w", encoding="utf-8", newline="")
         self.csv_writer = csv.writer(self.file_handle)
 
-    def process(self, data):
-        # data = np.atleast_1d(data) # Data shape may not be homogeneous
-        if not isinstance(data, Iterable):
-            data = [data]
-        self.csv_writer.writerow(data)
+    def process(self, data, cur_time):
+        row_data = [cur_time]
+        if isinstance(data, dict):
+            for value in data.values():
+                if isinstance(value, torch.Tensor):
+                    value = tensor_to_array(value).tolist()
```

Comment:
You `torch.tolist` for efficiency. Not that it is not necessary to do `cpu().detach()`, it should work whatever you specify as input, like magic!